examples/llama/tests/test_conversion.py

```diff
@@ -0,0 +1,251 @@
+
+
+def test_nt_to_hf_with_files(input_ids: torch.Tensor):
+    init_distributed(tp=1, dp=1, pp=1)(_test_nt_to_hf_with_files)(input_ids=input_ids, test_context=TestContext())
+
+
+def _test_hf_to_nt(parallel_context: ParallelContext, input_ids: torch.Tensor):
+    model_nt = create_nanotron_model(parallel_context)
+    model_hf = create_huggingface_model()
+    convert_hf_to_nt(model_hf, model_nt, CONFIG)
+    input_mask = torch.ones_like(input_ids)
+    logits_nt = model_nt.model(input_ids, input_mask).permute(1, 0, 2)
+    logits_hf = model_hf(input_ids).logits
+    assert logits_nt.size() == logits_hf.size()
+    assert torch.allclose(logits_nt, logits_hf, atol=ATOL), torch.mean(torch.abs(logits_nt - logits_hf))
+
+
+def test_hf_to_nt(input_ids: torch.Tensor):
+    init_distributed(tp=1, dp=1, pp=1)(_test_hf_to_nt)(input_ids=input_ids)
+
+
+def _test_hf_to_nt_with_files(parallel_context: ParallelContext, input_ids: torch.Tensor, test_context: TestContext):
+    # Create and save hf model.
+    model_hf = create_huggingface_model()
+    root = test_context.get_auto_remove_tmp_dir()
+    nt_path = root / "nanotron"
+    hf_path = root / "hf"
+    model_hf.save_pretrained(hf_path)
+    logits_hf = model_hf(input_ids).logits
+    del model_hf
+    # Perform conversion.
+    convert_hf_to_nt_and_save(hf_path, nt_path)
+    # Load nanotron and get logits.
+    input_mask = torch.ones_like(input_ids)
+    model_nt = load_nanotron_model(checkpoint_path=nt_path)
+    logits_nt = model_nt.model(input_ids, input_mask).permute(1, 0, 2)
+    assert logits_nt.size() == logits_hf.size()
+    assert torch.allclose(logits_nt, logits_hf, atol=ATOL)
+
+
+def test_hf_to_nt_with_files(input_ids: torch.Tensor):
+    init_distributed(tp=1, dp=1, pp=1)(_test_hf_to_nt_with_files)(input_ids=input_ids, test_context=TestContext())
+
+
+def _test_composed_conversion(parallel_context: ParallelContext):
+    # Get HF statedict.
+    model_hf = create_huggingface_model()
+    hf_sd = {key: val.clone() for key, val in model_hf.state_dict().items()}
+    # Convert once to nanotron, save its statedict.
+    model_nt = create_nanotron_model(parallel_context)
+    convert_hf_to_nt(model_hf, model_nt, CONFIG)
+    nt_sd = {key: val.clone() for key, val in model_nt.state_dict().items()}
+    # Convert back to HF, compare statedicts.
+    del model_hf
+    model_hf = create_huggingface_model()
+    convert_nt_to_hf(model_nt, model_hf, CONFIG)
+    hf_sd_new = model_hf.state_dict()
+    assert set(hf_sd_new) == set(hf_sd)
+    assert all(torch.all(hf_sd[key] == hf_sd_new[key]) for key in hf_sd_new)
+    # Convert to nanotron one more time, compare statedicts.
+    del model_nt
+    model_nt = create_nanotron_model(parallel_context)
+    convert_hf_to_nt(model_hf, model_nt, CONFIG)
+    nt_sd_new = model_nt.state_dict()
+    assert set(nt_sd_new) == set(nt_sd)
+    assert all(torch.all(nt_sd[key] == nt_sd_new[key]) for key in nt_sd_new)
+
+
+def test_composed_conversion():
```

Comment:
very nice test 🙌 !!

Reply:
@xrsrke thank you! (it was mainly written by @AleHD!)